docs/src/token-2022/onchain.md

```diff
@@ -230,6 +230,249 @@ do more work.
+program, most instructions involve multiple token types. If those token types are
+from different token programs, then our current implementation will fail.
+
+For example, if you want to swap tokens from the Token program for tokens from
+the Token-2022 program, then your program's instruction must provide each token
+program, so that your program may invoke them.
+
+Let's go through the steps to support both token programs in the same instruction.
+
+### Step 1: Update all instruction interfaces
+
+The first step is to update all instruction interfaces to accept a token program
+for each token type used in the program.
+
+For example, here is the previous definition for the `Swap` instruction:
+
+```rust
+///   Swap the tokens in the pool.
+///
+///   0. `[]` Token-swap
+///   1. `[]` swap authority
+///   2. `[]` user transfer authority
+///   3. `[writable]` token_(A|B) SOURCE Account, amount is transferable by user transfer authority,
+///   4. `[writable]` token_(A|B) Base Account to swap INTO.  Must be the SOURCE token.
+///   5. `[writable]` token_(A|B) Base Account to swap FROM.  Must be the DESTINATION token.
+///   6. `[writable]` token_(A|B) DESTINATION Account assigned to USER as the owner.
+///   7. `[writable]` Pool token mint, to generate trading fees
+///   8. `[writable]` Fee account, to receive trading fees
+///   9. `[]` Token program id
+///   10. `[optional, writable]` Host fee account to receive additional trading fees
+Swap {
+    pub amount_in: u64,
+    pub minimum_amount_out: u64
+}
+```
+
+`Swap` contains 3 different token types: token A, token B, and the pool token. Let's
+add a separate token program for each, transforming the instruction into:
+
+```rust
+///   Swap the tokens in the pool.
+///
+///   0. `[]` Token-swap
+///   1. `[]` swap authority
+///   2. `[]` user transfer authority
+///   3. `[writable]` token_(A|B) SOURCE Account, amount is transferable by user transfer authority,
+///   4. `[writable]` token_(A|B) Base Account to swap INTO.  Must be the SOURCE token.
+///   5. `[writable]` token_(A|B) Base Account to swap FROM.  Must be the DESTINATION token.
+///   6. `[writable]` token_(A|B) DESTINATION Account assigned to USER as the owner.
+///   7. `[writable]` Pool token mint, to generate trading fees
+///   8. `[writable]` Fee account, to receive trading fees
+///   9. `[]` Token (A|B) SOURCE program id
+///   10. `[]` Token (A|B) DESTINATION program id
+///   11. `[]` Pool Token program id
+///   12. `[optional, writable]` Host fee account to receive additional trading fees
+Swap {
+    pub amount_in: u64,
+    pub minimum_amount_out: u64
+}
+```
+
+Note the clarification on `9.`, and the new inputs of `10.` and `11.`.
+
+All of these additional accounts may make you wonder: how big will transactions
+get with these new accounts? If you are using both Token and Token-2022,
+the additional Token-2022 program pubkey will take up space in the transaction.
```

Comment:
Probably good to mention that is +32bytes for those that are not aware yet

Reply:
Makes sense, thanks!